src/Storages/ExternalStream/Kafka/WriteBufferFromKafka.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <atomic>
+#include <librdkafka/rdkafka.h>
+
+#include "IO/BufferWithOwnMemory.h"
+#include "IO/WriteBuffer.h"
+
+namespace DB
+{
+
+using RdKafkaTopicPtr = std::unique_ptr<rd_kafka_topic_t, void (*)(rd_kafka_topic_t *)>;
+
+class WriteBufferFromKafka : public BufferWithOwnMemory<WriteBuffer>
```

Comment:
If we don't want others to extend, it is is always a good idea to seal the class with `final`